tests/FakeItEasy.Specs/CallMatchingSpecs.cs

```diff
@@ -59,6 +59,61 @@ public interface IHaveAnObjectParameter
             int Bar(object arg);
         }
 
+        public interface IHaveAnIntParameter
+        {
+            int Bar(int arg);
+        }
+
+        [Scenario]
+        public static void Test(IHaveAnIntParameter fake, List<int> results, List<int> arguments)

```

Comment:
The name should be more descriptive.
Something like `ArgumentMatchersAreOnlyEvaluatedIfTheSpecifiedNumberOfTimesHaveBeenReached`
(yes, it's a mouthful... but at least we know what the test does)